persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stores/AbstractJdbcStore.java

```diff
@@ -0,0 +1,195 @@
+      this.log = getLog();
+   }
+
+   @Override
+   public void start() {
+      if (configuration.manageConnectionFactory()) {
+         ConnectionFactory factory = ConnectionFactory.getConnectionFactory(configuration.connectionFactory().connectionFactoryClass());
+         factory.start(configuration.connectionFactory(), factory.getClass().getClassLoader());
+         initializeConnectionFactory(factory);
+      }
+   }
+
+   @Override
+   public void stop() {
+      Throwable cause = null;
+      try {
+         tableManager.stop();
+      } catch (Throwable t) {
+         cause = t.getCause();
+         if (cause == null) cause = t;
+         log.debug("Exception while stopping", t);
+      }
+
+      try {
+         if (configuration.connectionFactory() instanceof ManagedConnectionFactory) {
+            log.tracef("Stopping mananged connection factory: %s", connectionFactory);
+            connectionFactory.stop();
+         }
+      } catch (Throwable t) {
+         if (cause == null) cause = t;
+         log.debug("Exception while stopping", t);
+      }
+      if (cause != null) {
+         throw new PersistenceException("Exceptions occurred while stopping store", cause);
+      }
+   }
+
+   @Override
+   public void clear() {
+      Connection conn = null;
+      PreparedStatement ps = null;
+      try {
+         String sql = tableManager.getDeleteAllRowsSql();
+         conn = connectionFactory.getConnection();
+         ps = conn.prepareStatement(sql);
+         int result = ps.executeUpdate();
+         if (log.isTraceEnabled()) {
+            log.tracef("Successfully removed %d rows.", result);
+         }
+      } catch (SQLException ex) {
+         log.failedClearingJdbcCacheStore(ex);
+         throw new PersistenceException("Failed clearing cache store", ex);
+      } finally {
+         JdbcUtil.safeClose(ps);
+         connectionFactory.releaseConnection(conn);
+      }
+   }
+
+   @Override
+   public void commit(Transaction tx) {
+      Connection connection;
+      try {
+         connection = getTxConnection(tx);
+         connection.commit();
+      } catch (SQLException e) {
+         log.sqlFailureTxCommit(e);
+         throw new PersistenceException(String.format("Error during commit of JDBC transaction (%s)", tx), e);
+      } finally {
+         destroyTxConnection(tx);
+      }
+   }
+
+   @Override
+   public void rollback(Transaction tx) {
+      Connection connection;
+      try {
+         connection = getTxConnection(tx);
+         connection.rollback();
+      } catch (SQLException e) {
+         log.sqlFailureTxRollback(e);
+         throw new PersistenceException(String.format("Error during rollback of JDBC transaction (%s)", tx), e);
+      } finally {
+         destroyTxConnection(tx);
+      }
+   }
+
+   protected Connection getTxConnection(Transaction tx) {
+      Connection connection = transactionConnectionMap.get(tx);
+      if (connection == null) {
```

Comment:
@wburns If ```connection == null```, then we actively get a new one. ```connectionFactory.getConnection()``` should never return null, instead it throws a PersistenceException if it is not possible to create a connection.